tcms/telemetry/static/telemetry/js/testing/status-matrix.js

```diff
@@ -0,0 +1,117 @@
+$(document).ready(() => {
+
+    loadInitialProduct();
+
+    document.getElementById('id_product').onchange = () => {
+        update_version_select_from_product();
+        update_build_select_from_product();
+        updateTestPlanSelectFromProduct();
+
+        redrawTable();
+    };
+
+    document.getElementById('id_version').onchange = redrawTable;
+    document.getElementById('id_build').onchange = redrawTable;
+    document.getElementById('id_test_plan').onchange = redrawTable;
+
+    $('#id_after').on('dp.change', redrawTable);
+    $('#id_before').on('dp.change', redrawTable);
+
+    drawTable();
+});
+
+function drawTable(query = {}) {
+    jsonRPC('Testing.status_matrix', query, data => {
+        const testRunIds = [];
+
+        data.test_runs.forEach(testRun => {
+            const testRunHeader = d3.select('table > thead > tr')
+                .append('td');
+
+            testRunHeader.append('a')
+                .text(testRun.summary)
+                .attr('href', `${window.location.origin}/plan/${testRun.run_id}/`);
+
+            testRunHeader.append('span')
+                .text(` ${testRun.start_date}`);
+
+            testRunIds.push(testRun.run_id);
+        });
+
+        data.test_cases.forEach(testCase => {
+            const testCaseRow = d3.select('table > tbody').append('tr');
+
+            testCaseRow.append('td').text(testCase.summary);
+
+            testRunIds.forEach(testRunId => {
+                const testExecution = data.test_executions.find(te => te.run_id === testRunId && te.case_id === testCase.case_id);
+
+                if (testExecution && testExecution.status !== 'IDLE') {
+                    let testExecutionCell = testCaseRow
+                        .append('td');
+
+                    if (testExecution.status === 'PASSED') {
+                        testExecutionCell.style('border-left', 'green 5px solid');
+                    } else {
+                        testExecutionCell.style('border-left', 'red 5px solid');
```

Comment:
color coding should come directly from the backend b/c this will break with translated statuses.